### PR TITLE
Add specific asset errors

### PR DESCRIPTION
**Pull request by thewilkybarkid**

### CODE REVIEW

File: tests/Workflow/MoveJatsAssetsTest.php [hunk 1]
```diff
@@ -561,4 +643,40 @@ public function it_uses_the_extension_if_an_octet_stream_content_type_is_returne
...
+
+        $marking = new Marking();
+        $task = new PutTask('service', ItemId::fromString('id'), ItemVersionNumber::fromInt(1), $document);
+        $transition = new Transition('transition', 'place1', 'place2');
+
+        $event = new Event($task, $marking, $transition);
+
+        $this->mock->save(
+            new Request('GET', 'http://origin-assets/assets/figure.foo'),
+            new Response(200, [], 'figure')
```
Comment (giorgiosironi) [inline tests/Workflow/MoveJatsAssetsTest.php]: can we make it a bit more clear that this is the `Response` that is at fault for not having a `Content-Type`? It gets lost in the rest of the setup but it's what makes this test different from the others

File: tests/Workflow/MoveJatsAssetsTest.php [hunk 2]
```diff
@@ -561,4 +643,40 @@ public function it_uses_the_extension_if_an_octet_stream_content_type_is_returne
...
+        );
+
+        $this->expectException(UnknownContentType::class);
+        $this->expectExceptionMessage('Unknown Content-Type');
+
+        $mover->onManipulate($event);
+    }
```
Comment (giorgiosironi) [inline tests/Workflow/MoveJatsAssetsTest.php]: only test missing here seems to be the one for an `InvalidContentType`

**Review by giorgiosironi — COMMENTED** [review]
Lacks some tests for EventListeners so far, if we intend to unit test them, plus a case for the Workflow class. Everything else fine

File: vendor-extra/JatsContentWorkflowBundle/composer.json [hunk 3]
```diff
@@ -1,14 +1,14 @@
 {
-    "name": "libero/jats-content-workflow",
-    "description": "JATS content workflow",
-    "type": "library",
+    "name": "libero/jats-content-workflow-bundle",
+    "description": "JATS content workflow bundle",
+    "type": "symfony-bundle",
```
Comment (thewilkybarkid) [inline vendor-extra/JatsContentWorkflowBundle/composer.json:4]: https://github.com/libero/content-store/pull/17#discussion_r301964287 didn't last long. 😅

File: composer.json [hunk 4]
```diff
@@ -11,12 +11,13 @@
...
         "psr-4": {
             "tests\\Libero\\ContentStore\\": "tests/",
-            "tests\\Libero\\JatsContentWorkflow\\": "vendor-extra/JatsContentWorkflow/tests/"
+            "tests\\Libero\\JatsContentWorkflowBundle\\": "vendor-extra/JatsContentWorkflowBundle/tests/"
         }
     },
     "require": {
         "php": "^7.3",
         "ext-ctype": "*",
+        "ext-intl": "*",
```
Comment (giorgiosironi) [inline composer.json:20]: not sure if it's the bundle or the application that should declare the dependency on the `intl` extension? (since the bundle has its own `composer.json`)

File: composer.json [hunk 5]
```diff
@@ -11,12 +11,13 @@
     "autoload-dev": {
         "psr-4": {
             "tests\\Libero\\ContentStore\\": "tests/",
-            "tests\\Libero\\JatsContentWorkflow\\": "vendor-extra/JatsContentWorkflow/tests/"
+            "tests\\Libero\\JatsContentWorkflowBundle\\": "vendor-extra/JatsContentWorkflowBundle/tests/"
```
Comment (giorgiosironi) [inline composer.json:14]: so much for unrelated changes? :sweat_smile:

File: composer.json (hunk 4 above)
Comment (thewilkybarkid) [reply composer.json:20]: The bundle depends on https://github.com/libero/content-store/blob/6235d9095e920b373c5e1e5b830d3977bf1329c7/vendor-extra/JatsContentWorkflowBundle/composer.json#L27 which the application replaces with the full extension.

File: composer.json (hunk 5 above)
Comment (thewilkybarkid) [reply composer.json:14]: https://github.com/libero/content-store/pull/16#discussion_r302076714 😳

File: vendor-extra/JatsContentWorkflowBundle/tests/Workflow/MoveJatsAssetsTest.php [hunk 6]
```diff
@@ -334,6 +338,84 @@ public function it_checks_the_origin_uri() : void
...
     }
 
+    /**
+     * @test
+     */
+    public function it_fails_if_the_asset_can_not_be_loaded() : void
+    {
+        $mover = new MoveJatsAssets('.+', 'http://public-assets/path', $this->filesystem, $this->client);
+
+        $document = FluentDOM::load(
```
Comment (giorgiosironi) [inline vendor-extra/JatsContentWorkflowBundle/tests/Workflow/MoveJatsAssetsTest.php:348]: any chance to extract `$document` and `$mock`'s expectations if/when they are repeated throughout the tests?

Comment (giorgiosironi) [reply vendor-extra/JatsContentWorkflowBundle/tests/Workflow/MoveJatsAssetsTest.php:348]: Though there is always some variation like the response code or a minor path.

**Review by giorgiosironi — APPROVED** [review]
